runtime/gc_structs/ClassArrayClassSlotIterator.cpp

```diff
@@ -36,32 +36,40 @@
  * @return the next slot containing an object reference
  * @return NULL if there are no more such slots
  */
-J9Class **
+J9Class *
 GC_ClassArrayClassSlotIterator::nextSlot() 
 {
-	J9Class **slotPtr;
+	J9Class *classPtr;
 
 	switch(_state) {
 	case classArrayClassSlotIterator_state_arrayClass:
-		slotPtr = (J9Class **)&_iterateClazz->arrayClass;
+		classPtr = (J9Class *)_iterateClazz->arrayClass;
 		if(!_isArrayClass) {
 			_state = classArrayClassSlotIterator_state_done;
 		} else {
 			_state += 1;
 		}
-		break;
+		if (NULL != classPtr) {
+			return classPtr;
+		}
+
 	case classArrayClassSlotIterator_state_componentType:
-		slotPtr = &((J9ArrayClass *)_iterateClazz)->componentType;
+		classPtr = ((J9ArrayClass *)_iterateClazz)->componentType;
 		_state += 1;
-		break;
+		if (NULL != classPtr) {
+			return classPtr;
+		}
+
 	case classArrayClassSlotIterator_state_leafComponentType:
-		slotPtr = &((J9ArrayClass *)_iterateClazz)->leafComponentType;
+		classPtr = ((J9ArrayClass *)_iterateClazz)->leafComponentType;
 		_state += 1;
-		break;
+		if (NULL != classPtr) {
+			return classPtr;
+		}
+
```

Comment:
Would you please add state `classArrayClassSlotIterator_state_done` here explicitly (currently it is handled as `default`). This does not change anything, just small clearing.